go/iot/arduino/protocol.txt

```diff
@@ -31,9 +34,18 @@ When the program starts, the following view would be registered.
         TagKeys: ["DeviceId", "SampleDate"]
 	ReportPeriod : 1
 
-Before the registration, the Raspberry Pi would make a negative response if there is any record request from Arduino.
+Before the registration, the Raspberry Pi would not make any response to the the request from Arduino.
+
+After the registration, the Raspberry Pi would make a negative or positive response to any record request from Arduino.
+
+When the Pi would make a negative response:
+    1. The measure doesn't bind to any views.
+    2. The tag key doesn't bind to any views.
+    2. Pi could not parse the encapsulated data because of the format error.
+    3. The number of tag keys is not equal to the number of tag values.
 
-After the registration, the Raspberrry Pi would make a negative or positive response to any record request from Arduino.
+When the Pi successfully parses and records data, it would make a positive response to the request. Note that it would
+still responds positively if there is something wrong with the OpenCensus framework such as networking failure.
```

Comment:
Can you elaborate more on this? By "networking failure" I would assume it's Pi cannot connect to Stackdriver backend, why respond positively in this case?